cyral/data_source_cyral_integration_idp.go

```diff
@@ -0,0 +1,145 @@
+package cyral
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/cyralinc/terraform-provider-cyral/client"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type IdPIntegrations struct {
+	Connections *Connections `json:"connections,omitempty"`
+}
+
+type Connections struct {
+	Connections []*Connection `json:"connections,omitempty"`
+}
+
+type Connection struct {
+	DisplayName            string `json:"displayName,omitempty"`
+	Alias                  string `json:"alias,omitempty"`
+	SingleSignOnServiceURL string `json:"singleSignOnServiceURL,omitempty"`
+	Enabled                bool   `json:"enabled,omitempty"`
+}
+
+func dataSourceIntegrationIdP() *schema.Resource {
+	return &schema.Resource{
+		Description: "Retrieve and filter IdP integrations.",
+		ReadContext: dataSourceIntegrationIdPRead,
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Description: "Filter results by the name of an existing IdP integration.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			"type": {
+				Description: "Filter results by the IdP integration type.",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"aad",
+					"adfs",
+					"forgerock",
+					"gsuite",
+					"okta",
+					"ping_one",
```

Comment:
The alias of Ping One integrations are prefixed by `pingone` instead, changing this should be enough to fix filtering by this type of IdP.
```suggestion
					"pingone",
```